ts/parsers/paragraphparser.ts

```diff
@@ -1,37 +1,155 @@
+        const level = getValueAtPath(paragraph, '["a:pPr"][0]["$"]["lvl"]');
+
+        return level ? parseInt(level) : 0;
+    }
+
+    public static getListType(paragraph): ListType {
+        if (checkPath(paragraph, '["a:pPr"][0]["a:buAutoNum"]')) {
+            return ListType.Ordered;
+        }
+
+        return ListType.UnOrdered;
+    }
+
+    public static restructureList(list: List): List {
+        for (let i = 0; i < list.listItems.length - 1; i++) {
+            if (list.listItems[i + 1].list) {
+                list.listItems[i]["list"] = this.restructureList(list.listItems[i + 1].list);
+                list.listItems.splice(i + 1, 1);
+            }
+        }
+        return list;
+    }
+
     public static extractParagraphElements(paragraphs: any[]): PowerpointElement["paragraph"] {
         if (!paragraphs || paragraphs.length === 0) {
             return null;
         }
 
-        return paragraphs.map((paragraph) => {
-            const textElements = paragraph["a:r"] || [];
-            const content = textElements.map((txtElement) => {
-                return {
-                    text: txtElement["a:t"] || "",
-                    textCharacterProperties: this.determineTextProperties(getValueAtPath(txtElement, '["a:rPr"][0]'))
-                };
-            });
+        const paras = [];
+        const stack = [];
+        const paragraph: Paragraph = {
+            list: {
+                listType: ListType.Ordered,
+                listItems: []
+            }
+        };
+        let currentParagraph = paragraph;
+        let currentLevel = -1;
 
-            return {
-                content: content,
-                paragraphProperties: this.determineParagraphProperties(paragraph)
-            };
-        });
+        for (const p of paragraphs) {
+            if (this.isList(p)) {
+                const listLevel = this.getListlevel(p);
+
+                //if its the first of the list kind
+                if (currentLevel === -1) {
+                    currentLevel = 0;
+                    currentParagraph.list.listType = this.getListType(p);
+                    currentParagraph.list.listItems.push(this.getParagraph(p));
+                    stack.push(currentParagraph);
+                }
+                //if the level is same keep pushing the list items in the same array
+                else if (listLevel === currentLevel) {
+                    currentParagraph.list.listItems.push(this.getParagraph(p));
+                } else if (listLevel > currentLevel) {
+                    //edge case to handle if multiple levels are jumped ahead
+                    //create empty paragraphs/lists to maintain hierarchy and fill in the level gaps
+                    while (currentLevel < listLevel - 1) {
+                        const emptyPara: Paragraph = {
+                            list: {
+                                listType: ListType.UnOrdered,
```

Comment:
I guess you are yet to implement the actual list type for the child you are trying to insert here

Reply:
Me and Ryasath agreed to put list type as none, but that doesn't work out for me at this point, as i don't know if this list would be populated with the items in the next iteration. 
@kryasaleg already was working on the implementation of skipping on rendering the list bullet items or numbers, for which it doesn't have a content.
however i could reiterate the list and set the properties accordingly as well. but can't be done at this level

Reply:
I don't see a problem with using the the list type for the child that you will eventually add under this series of parents. I'll discuss this

Reply:
i hope its clear now after our discussion. 
but will still put it on hold until confirmed by @kryasaleg, if he is okay with it. If there are any issues on the renderer side then i will re iterate the lists, and change the `listType` to 'none' wherever applicable. 
But Ryasath bhai already handled that i guess so i will leave it as is for now.

Reply:
It it clear thanks Rabi, I'll raise it with Ryasath and if he doesn't see an issue then I'm happy!